### PR TITLE
feat(provider): add Mistral provider

### PR DESCRIPTION
**Pull request by agpituk**
## Summary

Add Mistral provider implementation using the OpenAI-compatible base with custom handling for message patching and unsupported API fields.

## Changes

- Add `providers/mistral/mistral.go` - Provider implementation
- Add `providers/mistral/mistral_test.go` - Comprehensive test suite
- Update `internal/testutil/fixtures.go` - Add Mistral model mappings
- Update `docs/providers.md` - Add Mistral documentation

### Features

- Chat completion and streaming
- Tool calling support
- Reasoning support (Magistral models)
- Vision support (Pixtral models)
- Embeddings (mistral-embed)
- List models

### Mistral API Quirks

Mistral's OpenAI-compatible API has a few differences that require preprocessing:

1. **Tool-to-user message patching** - Mistral requires an assistant message between a tool result and the next user message. The provider automatically inserts an `"OK"` assistant message in these sequences.
2. **User field unsupported** - The `user` field is stripped before API calls.
3. **Reasoning effort unsupported** - Mistral doesn't accept the `reasoning_effort` parameter (Magistral models reason automatically). The field is stripped to prevent 422 errors.

### Unit Tests
- Provider creation and API key validation
- Capabilities reporting
- `preprocessParams` (user stripping, reasoning effort stripping, message patching, param preservation)
- `patchMessages` (no-op, insertion, multiple sequences, edge cases, immutability)

### Integration Tests
- Basic completion and streaming
- System messages and conversations
- Tool calling
- Agent loops with weather and calculator tools
- Reasoning (Magistral)
- Embeddings

```bash
# Run unit tests
go test -v ./providers/mistral/... -run 'Test[^I]'

# Run integration tests (requires MISTRAL_API_KEY)
go test -v ./providers/mistral/... -run 'TestIntegration'
```

All tests passing:
```
ok      github.com/mozilla-ai/any-llm-go/providers/mistral    2.457s
```

## Checklist

- [x] Tests pass locally (`go test -short ./...`)
- [x] Linting passes (`golangci-lint run ./...`)
- [x] Documentation updated (if needed)
- [x] No breaking changes (or documented in summary)

### CODE REVIEW

File: providers/mistral/mistral_test.go [hunk 1]
```diff
@@ -0,0 +1,641 @@
...
+	ctx := context.Background()
+	resp, err := provider.Embedding(ctx, providers.EmbeddingParams{
+		Model: testutil.EmbeddingModel(providerName),
+		Input: "Hello, world!",
+	})
+	require.NoError(t, err)
+
+	require.NotEmpty(t, resp.Data)
+	require.NotEmpty(t, resp.Data[0].Embedding)
+	require.Greater(t, len(resp.Data[0].Embedding), 0)
```
Comment (peteski22) [inline providers/mistral/mistral_test.go]:
```suggestion
```

I don't think you need this as the line above checks it's not empty

**Review by peteski22 — APPROVED** [review]
